packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -403,13 +403,51 @@ gmaps.PolylineOptions _polylineOptionsFromPolyline(
   final List<gmaps.LatLng> paths =
       polyline.points.map(_latLngToGmLatLng).toList();
 
+  final bool hasPattern = polyline.patterns.isNotEmpty;
+
+  final List<gmaps.IconSequence> iconSequences = <gmaps.IconSequence>[];
+
+  if (hasPattern) {
+    List<Object>? getPattern(List<List<Object>> input, String patternName) {
+      final List<Object> found = input.firstWhere(
+        (List<Object> innerList) =>
+            innerList.isNotEmpty &&
+            innerList[0] is String &&
+            innerList[0] == patternName,
+        orElse: () => <Object>[],
+      );
+      return found.isEmpty ? null : found;
+    }
+
+    final List<List<Object>> patternJson = polyline.patterns
+        .map((PatternItem element) => element.toJson() as List<Object>)
```

Comment:
/cc @stuartmorgan this needs to be done because `PatternItem` is an opaque, json-only backed object. If we cleaned this up a little bit, this PR would be slightly simpler.

(This is not a huge issue because normally the list of PatternItems is tiny.)